src/SqlPersistence/ScriptLocation.cs

```diff
@@ -20,20 +19,15 @@ static string GetCurrentDirectory(ReadOnlySettings settings)
         {
             return scriptDirectory;
         }
-        var entryAssembly = Assembly.GetEntryAssembly();
-        if (entryAssembly == null)
+
+        var baseDir = AppDomain.CurrentDomain.BaseDirectory;
```

Comment:
So this change likely works, but it's still relying on something that can have some edge cases. `AppDomain.CurrentDomain.BaseDirectory` is set by the thing the created the `AppDomain`, so we're relying on the fact that it's been set correctly in all scenarios now. Probably safe generally, but it's certainly possible to make it fail.